modules/service/src/main/scala/lucuma/odb/service/ProgramService.scala

```diff
@@ -162,13 +179,60 @@ object ProgramService {
           }
 
         // Update proposals. This can fail in a few ways.
-        val updateProposals: F[List[Program.Id]] =
-          SET.proposal.fold(Nil.pure[F]) {
-            proposalService.updateProposals(_)
+        val updateProposals: F[Either[UpdateProgramsError, List[Program.Id]]] =
```

Comment:
At some point we should really change this to return `F[Result[List[Program.Id]]]` since all we do with the structured error is turn it into a `Problem` up in the mutation. It wasn't initially clear what kind of values the services needed to compute.

Reply:
Ah. OK. I'll change that. I had initially started in that direction, but changed when the services I was using for inspiration weren't return `Result`s. I thought there might be some reason for separating the types that was beyond my ken.

Reply:
Originally I had wanted to keep the service layer independent of the GraphQL layer (`Result` is a Grackle data type) but it resulted in a lot of indirection that was probably worse than just using `Result`. I don't have strong feelings about it anymore.